### PR TITLE
test(eth-json-rpc-middleware): Round out test suite

### PR DESCRIPTION
**Pull request by rekmarks**
## Explanation

Adds a number of missing unit test cases to `eth-json-rpc-middleware`, in preparation for migrating the package to `JsonRpcEngineV2`. Opportunistically rearranges some test utilities, e.g. renames `build*` functions to `create*`.

> [!TIP]
> I recommend reviewers to hide whitespace.

## References

N/A

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've communicated my changes to consumers by [updating changelogs for packages I've changed](https://github.com/MetaMask/core/tree/main/docs/contributing.md#updating-changelogs), highlighting breaking changes as necessary
- [x] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

---

> [!NOTE]
> Adds extensive tests across eth-json-rpc-middleware, refactors test helpers, minor middleware typings/docs tweaks, and raises coverage thresholds.
> 
> - **Tests (eth-json-rpc-middleware)**:
>   - Add comprehensive unit tests for `block-cache`, `block-ref`, `block-ref-rewrite`, `block-tracker-inspector`, `retryOnEmpty`, `fetch`, `inflight-cache`, `wallet`, and package `index` exports.
>   - Cover edge cases: skip-cache flags, non-cacheable methods, pending/latest/hex block tags, empty results, transaction/receipt caching conditions, retry exhaustion, execution-reverted errors, and block tracker update logic.
> - **Test Utilities**:
>   - Introduce `createRequest`, `createProviderAndBlockTracker`, `createEngine`, `createFinalMiddlewareWithDefaultResult`, `createSimpleFinalMiddleware`, and stub helpers; rename `build*` helpers to `create*`.
>   - Remove `test/util/createHitTrackerMiddleware.ts`.
> - **Middleware/Internal**:
>   - Minor refactors and JSDoc/typing improvements in `block-ref-rewrite.ts`, `block-tracker-inspector.ts`, and `retryOnEmpty.ts` (no functional changes intended).
> - **Config**:
>   - Increase Jest coverage thresholds in `jest.config.js` and include `./test` in `tsconfig.json`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 24a3ce9ba8844770e1be0e724dcd2eabe5c88f11. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/eth-json-rpc-middleware/src/block-tracker-inspector.ts [hunk 1]
```diff
@@ -10,78 +11,42 @@ import type {
...
+ * @param options - The options for the middleware.
+ * @param options.blockTracker - The block tracker to use.
+ * @returns The middleware.
  */
-function hasProperty<ObjectToCheck, Property extends ValidPropertyType>(
-  objectToCheck: ObjectToCheck,
-  property: Property,
-): objectToCheck is ObjectToCheck & Record<Property, unknown> {
-  return Object.hasOwnProperty.call(objectToCheck, property);
-}
```
Comment (rekmarks) [inline packages/eth-json-rpc-middleware/src/block-tracker-inspector.ts:39]: Now imported from `@metamask/utils`.

File: packages/eth-json-rpc-middleware/src/block-cache.test.ts [hunk 2]
```diff
@@ -1,57 +1,373 @@
...
-  const provider = providerFromEngine(engine);
+describe('block cache middleware', () => {
+  let provider: ReturnType<typeof createProviderAndBlockTracker>['provider'];
+  let blockTracker: ReturnType<
+    typeof createProviderAndBlockTracker
+  >['blockTracker'];
 
-  const blockTracker = new PollingBlockTracker({
-    provider,
+  beforeEach(() => {
```
Comment (mcmire) [inline packages/eth-json-rpc-middleware/src/block-cache.test.ts:17]: Any reason for preferring `beforeEach` over factory functions in tests? In my experience, they serve as a sort of junk drawer for setup data that has a tendency to get more and more disorganized over time, and they make it more difficult to write tests that extend the setup data in some way. So I tend to avoid hooks completely (except for one-offs like enabling/disabling fake timers). But I also recognize that there are some advantages to using them, so I'm curious to know how you view them.

File: packages/eth-json-rpc-middleware/src/wallet.test.ts [hunk 3]
```diff
@@ -20,17 +20,17 @@ const testTxHash =
 const testMsgSig =
   '0x68dc980608bceb5f99f691e62c32caccaee05317309015e9454eba1a14c3cd4505d1dd098b8339801239c9bcaac3c4df95569dcf307108b92f68711379be14d81c';
 
-function createTestSetup() {
+const createProviderAndEngine = () => {
```
Comment (mcmire) [inline packages/eth-json-rpc-middleware/src/wallet.test.ts]: Do we always need the `provider`? Curious if in all of these middleware tests we can create the provider only when we need it.

File: packages/eth-json-rpc-middleware/test/util/helpers.ts [hunk 4]
```diff
@@ -1,9 +1,31 @@
...
+>(
+  request: Input,
+): Output => {
+  return {
+    jsonrpc: '2.0',
+    id: request.id ?? '1',
+    method: request.method ?? 'test_request',
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    params: request.params === undefined ? [] : request.params,
+  } as Output;
```
Comment (mcmire) [inline packages/eth-json-rpc-middleware/test/util/helpers.ts:26]: Why are we forcing `params` to be present and why are we forcing the return type to be `Output`? Is this here to clean up some things but it would take a larger refactor to truly fix this?

File: packages/eth-json-rpc-middleware/src/block-cache.test.ts (hunk 2 above)
Comment (rekmarks) [reply packages/eth-json-rpc-middleware/src/block-cache.test.ts:17]: I generally agree with you, except in the case where there's cleanup to perform, in this case `blockTracker.destroy()`. In such cases I generally find the hooks to be worth it.

File: packages/eth-json-rpc-middleware/src/wallet.test.ts (hunk 3 above)
Comment (rekmarks) [reply packages/eth-json-rpc-middleware/src/wallet.test.ts]: We never used it. Removed.

File: packages/eth-json-rpc-middleware/test/util/helpers.ts (hunk 4 above)
Comment (rekmarks) [reply packages/eth-json-rpc-middleware/test/util/helpers.ts:26]: Re: `params`, `eth_*` JSON-RPC method parameters are conventionally always arrays, and an empty array is equivalent to no params. It's more succinct to default to something than to omit the field, so here we are.

Re: casting the return value to be `Output`, we need the Output to be a `JsonRpcRequest` and that's what we're doing here. That said, I did make the params `Json[]`, which is more accurate.

**Review by mcmire — APPROVED** [review]
LGTM.
